src/main/java/hudson/plugins/ec2/ssh/EC2UnixLauncher.java

```diff
@@ -278,6 +323,19 @@ public boolean verifyServerHostKey(String hostname, int port, String serverHostK
         }
     }
 
+    private String getEC2HostName(EC2Computer computer, Instance inst) {
+        if (computer.getNode().usePrivateDnsName) {
+            return inst.getPrivateDnsName();
+        } else {
+            String host = inst.getPublicDnsName();
+            // If we fail to get a public DNS name, use the private IP.
+            if (host == null || host.equals("")) {
+                host = inst.getPrivateIpAddress();
```

Comment:
@KostyaSha Not sure actually ... that code was already there. I just refactored it out into a method. Let me know what way you think it should be.